tfjs-backend-webgpu/src/kernels/BatchMatMul_impl.ts

```diff
@@ -94,7 +94,7 @@ export function batchMatMulImpl({
                    (outerShapeA % 4 === 0 && transposeA)) &&
       outerShapeB % 4 === 0 && !transposeB;
   let program: WebGPUProgram;
-  if (outerShapeA * outerShapeB <= 32) {
```

Comment:
From the point of global memory load, `MatMulReduceProgram` is similar as a naive matmul method, the count of global memory load is 2*M*K*N. For the other shared memory version of matmul, the count is M*K*N*(1/tileColSize + 1/tileRowSize). For small size matmul, GPU bandwidth will be enough and no bottleneck happens when using  `MatMulReduceProgram`.  I think that different hardwares have its own critical value, so we need to test on more hardwares and find a conservative value for the condition.
And [2, 2048, 64] is not friend to other matmul versions, but [8, 2048, 16] is very friend, so you should test the cases, and compare their performance.

Reply:
Below is the test result with [8, 2048, 16]
```
CFL
Matmul[8, 2048, 16]: 0.133998 ms, MatMulReduceProgram: 0.133998
Matmul[8, 2048, 16]: 0.134915 ms, MatMulReduceProgram: 0.134915
Matmul[8, 2048, 16]: 0.266079 ms, MatMulSmallOutputSizeProgram: 0.266079
Matmul[8, 2048, 16]: 0.284662 ms, MatMulSmallOutputSizeProgram: 0.284662
TGL
Matmul[8, 2048, 16]: 0.056562 ms, MatMulReduceProgram: 0.056562
Matmul[8, 2048, 16]: 0.055417 ms, MatMulReduceProgram: 0.055417
Matmul[8, 2048, 16]: 0.281092 ms, MatMulSmallOutputSizeProgram: 0.281092
Matmul[8, 2048, 16]: 0.280988 ms, MatMulSmallOutputSizeProgram: 0.280988
```
You can see that `MatMulReduceProgram` is still faster than current default matmul pass.